server/render/index.js

```diff
@@ -122,15 +122,36 @@ export function render( element, key = JSON.stringify( element ), req ) {
 	//todo: render an error?
 }
 
+/**
+ * Check a query object against an array of whitelisted keys.
+ *
+ * If any key in the query is not present in the whitelist, it is not cacheable.
+ *
+ * @param  {?Object}        query                Query object
+ * @param  {?Array<string>} whitelistedQueryKeys Whitelisted keys
+ * @return {boolean}                             True if all query keys are whitelisted
+ */
+export function isCacheableQuery( query = null, whitelistedQueryKeys = null ) {
+	if ( isEmpty( query ) ) {
+		return true;
+	}
+	return every( Object.keys( query ), key => includes( whitelistedQueryKeys, key ) );
+}
+
 export function serverRender( req, res ) {
 	const context = req.context;
+
 	let title,
 		metas = [],
 		links = [],
 		cacheKey = false;
 
-	if ( isSectionIsomorphic( context.store.getState() ) && ! context.user ) {
-		cacheKey = getCacheKey( context );
+	if (
+		isSectionIsomorphic( context.store.getState() ) &&
+		! context.user &&
```

Comment:
I'm seeing SSR when logged in. Is this misbehaving?

Reply:
Dev environment? That's a known issue -- the ol' `calypso.localhost`-can't-see-the-wordpress-dot-com-auth-cookie-and-believes-we're-logged-out.